src/jit/emitxarch.cpp

```diff
@@ -2588,15 +2133,76 @@ void emitter::emitIns_Nop(unsigned size)
     emitCurIGsize += size;
 }
 
+void emitter::emitPredictInstSize(instrDesc* id)
+{
+    emitPredictInstSize(id, 0, false, true);
+}
+
+void emitter::emitPredictInstSize(instrDesc* id, UNATIVE_OFFSET prevSize, bool skipNewSize, bool skipPrevSize)
```

Comment:
This approach to either fully or partially recomputing the size seems confusing and error-prone.